src/main/java/com/expenses/ExpenseService.java

```diff
@@ -0,0 +1,62 @@
+package com.expenses;
+
+import java.time.LocalDate;
+import java.util.*;
+
+public class ExpenseService {
+   private Set<Expense> expenses = new HashSet<>();
+
+    public void addExpense(Expense expense)  {
+         expenses.add(expense);
+   }
+
+    public Set<Expense> findExpensesInRange(LocalDate from, LocalDate to){
+        Set<Expense> expensesInRange = new HashSet<>();
+       for (Expense expense:expenses) {
+           if ((expense.getDate().isAfter(from) | expense.getDate().isEqual(from)) && (expense.getDate().isBefore(to)|expense.getDate().isEqual(to))){
```

Comment:
Całkiem zręcznie zrobione, będzie działało co widać w testach.

Mam dwie uwagi:
1. Wewnątrz do operacji `OR` użyłeś operatora `|` zamiast `||`. W tej sytuacji zadziała podobnie, ale warto zdawać sobie sprawę z poważnych różnic pomiędzy `|` oraz `||`. Odsyłam do internetu po informacje. :)
2. Warunki `(expense.getDate(isAfter(from)) || expense.getDate().isEqual(from))` można zamienić na jedno, prostsze sprawdzenie. By trochę naprowadzić na rozwiązanie, pomyśl o sprawdzeniu przypadku `czy x jest mniejszy bądź równy 0`. Podchodząc dosłownie można sprawdzić `(czy x jest mniejszy od 0) LUB (czy x jest równy 0)`. To zadziała tak jak powyższy kod. Niemniej, można to uprościć sprawdzając `(czy x nie jest większy od 0)`. No bo jak nie jest większy to na pewno jest albo równy albo mniejszy. :)